<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Estimating Linkage Disequilibrium within a multi-ethnic cohort • gauss</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Estimating Linkage Disequilibrium within a multi-ethnic cohort">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="default" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">gauss</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../index.html">Get Started</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/afmix_example.html">Estimating ancestry proportions</a></li>
    <li><a class="dropdown-item" href="../articles/computeLD_example.html">Estimating linkage disequilibrium</a></li>
    <li><a class="dropdown-item" href="../articles/dist_example.html">Imputing Z-scores of unmeasured SNPs</a></li>
    <li><a class="dropdown-item" href="../articles/jepeg_example.html">Transcriptome wide association study</a></li>
    <li><a class="dropdown-item" href="../articles/fiqt_example.html">Winner's curse adjustment</a></li>
  </ul>
</li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-reference-panels" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Reference Panels</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-reference-panels">
<li><a class="dropdown-item" href="../articles/ref_33KG.html">33KG Panel</a></li>
    <li><a class="dropdown-item" href="../articles/ref_1KG.html">1KG Panel</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Functions</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><a class="nav-link" id="pkgdown-search-button" href="../#"><span class="fa fa-search"></span></a></li>
<li class="nav-item"><a class="nav-link" href="https://github.com/statsleelab/gauss"><span class="fa fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Estimating Linkage Disequilibrium within a multi-ethnic cohort</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/statsleelab/gauss/blob/HEAD/vignettes/computeLD_example.Rmd"><code>vignettes/computeLD_example.Rmd</code></a></small>
      <div class="d-none name"><code>computeLD_example.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Linkage disequilibrium (LD) quantifies the degree of non-random
association between alleles at different loci within a population. The
estimation of LD in multi-ethnic cohorts, however, can be challenging
due to the underlying population structure. The <code><a href="../reference/computeLD.html">computeLD()</a></code>
function in the GAUSS package addresses this challenge by offering an
efficient and accurate method to estimate LD, factoring in the mixed
ancestries present in the cohort. This vignette will provide you a
step-by-step guide on using <code><a href="../reference/computeLD.html">computeLD()</a></code> for estimating LD
for SNPs within a multi-ethnic Schizophrenia GWAS cohort.</p>
</div>
<div class="section level2">
<h2 id="load-necessary-packages">Load necessary packages<a class="anchor" aria-label="anchor" href="#load-necessary-packages"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load necessary packages</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/statsleelab/gauss">gauss</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/hadley/reshape" class="external-link">reshape2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r-datatable.com" class="external-link">data.table</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://haozhu233.github.io/kableExtra/" class="external-link">kableExtra</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="overview-of-computeld">Overview of computeLD()<a class="anchor" aria-label="anchor" href="#overview-of-computeld"></a>
</h2>
<p>The <code><a href="../reference/computeLD.html">computeLD()</a></code> function in the GAUSS package is
designed to estimate Linkage Disequilibrium (LD) by leveraging the
observed or estimated ancestry proportions of a target cohort. It
calculates LD for specific genomic regions as a weighted sum of LD
values across various ancestries. The LD values for each ancestry are
estimated from a reference panel. In this example, we utilize <a href="https://statsleelab.github.io/gauss/articles/ref_33KG.html" class="external-link">the
33KG reference panel</a> for our estimations.</p>
<p>The function requires following arguments:</p>
<ul>
<li>
<code>chr</code>: Chromosome number.</li>
<li>
<code>start_bp</code>: Starting base pair position of the estimation
window.</li>
<li>
<code>end_bp</code>: Ending base pair position of the estimation
window.</li>
<li>
<code>pop_wgt_df</code>: R data frame containing population IDs and
wights</li>
<li>
<code>input_file</code>: File name of the association Z-score
data</li>
<li>
<code>reference_index_file</code>: File name of reference panel
index data</li>
<li>
<code>reference_data_file</code>: File name of reference panel
data</li>
<li>
<code>reference_pop_desc_file</code>: File name of reference panel
population description data</li>
<li>
<code>af1_cutff</code>: Cutoff of reference allele (a1)
frequency</li>
</ul>
<p>The <code><a href="../reference/computeLD.html">computeLD()</a></code> function returns a list with two
elements:</p>
<ul>
<li>
<code>snplist</code>: A data frame containing information on the
SNPs used in the LD estimation.</li>
<li>
<code>cormat</code>: A matrix of LD estimates for each pair of
SNPs.</li>
</ul>
</div>
<div class="section level2">
<h2 id="preparing-the-input-data">Preparing the Input Data<a class="anchor" aria-label="anchor" href="#preparing-the-input-data"></a>
</h2>
<div class="section level3">
<h3 id="ancestry-proportion-data">Ancestry Proportion Data<a class="anchor" aria-label="anchor" href="#ancestry-proportion-data"></a>
</h3>
<p>The ancestry proportion data <code>pop_wgt_df</code> should include
two columns with the following names: pop (population abbreviation) wgt
(weight or proportion). This information should be estimated using the
<code><a href="../reference/afmix.html">afmix()</a></code> function as described <a href="https://statsleelab.github.io/gauss/articles/afmix_example.html" class="external-link">here</a>.</p>
<p>Here, we load the ancestry proportion data generated from the <a href="https://statsleelab.github.io/gauss/articles/afmix_example.html" class="external-link">vignette</a>
for <code><a href="../reference/afmix.html">afmix()</a></code> function:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the ancestry proportion data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"PGC2_SCZ_ANC_Prop"</span><span class="op">)</span> <span class="co"># data frame name: PGC2_SCZ_ANC_Prop</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">PGC2_SCZ_ANC_Prop</span><span class="op">)</span></span>
<span><span class="co">#&gt;   pop   wgt</span></span>
<span><span class="co">#&gt; 1 ACB 0.006</span></span>
<span><span class="co">#&gt; 2 ASW 0.036</span></span>
<span><span class="co">#&gt; 3 BEB 0.005</span></span>
<span><span class="co">#&gt; 4 CCE 0.008</span></span>
<span><span class="co">#&gt; 5 CCS 0.004</span></span>
<span><span class="co">#&gt; 6 CDX 0.018</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="association-z-score-data">Association Z-score Data<a class="anchor" aria-label="anchor" href="#association-z-score-data"></a>
</h3>
<p>The association Z-score data file should be a space-delimited text
file including six columns with the following names: <code>rsid</code>
(SNP ID), <code>chr</code> (chromosome number), <code>bp</code> (base
pair position), <code>a1</code> (reference allele), <code>a2</code>
(alternative allele), and <code>z</code> (association Z-score).</p>
<p>We assign the path to the association Z-score data file here:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Define the path to the input file</span></span>
<span><span class="va">input_file</span> <span class="op">&lt;-</span> <span class="st">"../data/PGC2_3Mb.txt"</span></span>
<span></span>
<span><span class="co"># Input file should include six columns (rsid, chr, bp, a1, a2, and z)</span></span>
<span><span class="va">input_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="va">input_file</span>, header <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">input_data</span><span class="op">)</span></span>
<span><span class="co">#&gt;          rsid   chr        bp     a1     a2         z</span></span>
<span><span class="co">#&gt;        &lt;char&gt; &lt;int&gt;     &lt;int&gt; &lt;char&gt; &lt;char&gt;     &lt;num&gt;</span></span>
<span><span class="co">#&gt; 1:  rs1004467    10 104594507      A      G  6.686674</span></span>
<span><span class="co">#&gt; 2:  rs1008013    10 103548866      A      T -1.769923</span></span>
<span><span class="co">#&gt; 3: rs10128116    10 103717613      A      G -1.883298</span></span>
<span><span class="co">#&gt; 4:  rs1015037    10 105547517      T      G -1.917614</span></span>
<span><span class="co">#&gt; 5: rs10159775    10 103184297      A      G  1.304979</span></span>
<span><span class="co">#&gt; 6: rs10159838    10 105473937      A      G  2.582526</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="reference-panel-data-files">Reference Panel Data Files<a class="anchor" aria-label="anchor" href="#reference-panel-data-files"></a>
</h3>
<p>Next, we assign the paths to reference panel data files:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Paths to the reference files (replace these with your actual paths)</span></span>
<span><span class="va">reference_index_file</span> <span class="op">&lt;-</span><span class="st">"../ref/33KG/33kg_index.gz"</span></span>
<span><span class="va">reference_data_file</span> <span class="op">&lt;-</span> <span class="st">"../ref/33KG/33kg_geno.gz"</span></span>
<span><span class="va">reference_pop_desc_file</span><span class="op">&lt;-</span><span class="st">"../ref/33KG/33kg_pop_desc.txt"</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="running-computeld">Running computeLD()<a class="anchor" aria-label="anchor" href="#running-computeld"></a>
</h2>
<p>With the necessary arguments and data prepared, we can now estimate
the Linkage Disequilibrium for a specific genomic region using the
<code><a href="../reference/computeLD.html">computeLD()</a></code> function. For this example, we focus on a 1Mb
genomic region located at Chromosome 10: 104-105 Mb.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">af1_cutoff</span> <span class="op">=</span> <span class="fl">0.001</span></span>
<span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/computeLD.html">computeLD</a></span><span class="op">(</span>chr<span class="op">=</span><span class="fl">10</span>, </span>
<span>                 start_bp <span class="op">=</span> <span class="fl">104000001</span>, </span>
<span>                 end_bp <span class="op">=</span> <span class="fl">105000000</span>,</span>
<span>                 pop_wgt_df <span class="op">=</span> <span class="va">PGC2_SCZ_ANC_Prop</span>,</span>
<span>                 input_file <span class="op">=</span> <span class="va">input_file</span>, </span>
<span>                 reference_index_file <span class="op">=</span> <span class="va">reference_index_file</span>,</span>
<span>                 reference_data_file <span class="op">=</span> <span class="va">reference_data_file</span>, </span>
<span>                 reference_pop_desc_file <span class="op">=</span> <span class="va">reference_pop_desc_file</span>,</span>
<span>                 af1_cutoff <span class="op">=</span> <span class="va">af1_cutoff</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="examining-the-results">Examining the Results<a class="anchor" aria-label="anchor" href="#examining-the-results"></a>
</h2>
<p>Let’s display the first few entries of the results:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">snplist</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="st">"html"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr>
<th style="text-align:left;">
rsid
</th>
<th style="text-align:right;">
chr
</th>
<th style="text-align:right;">
bp
</th>
<th style="text-align:left;">
a1
</th>
<th style="text-align:left;">
a2
</th>
<th style="text-align:right;">
af1mix
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
rs3758549
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
104004195
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:right;">
0.1928059
</td>
</tr>
<tr>
<td style="text-align:left;">
rs1541046
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
104005386
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:right;">
0.6625196
</td>
</tr>
<tr>
<td style="text-align:left;">
rs2296887
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
104005410
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
T
</td>
<td style="text-align:right;">
0.1591055
</td>
</tr>
<tr>
<td style="text-align:left;">
rs10748818
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
104015279
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:right;">
0.1664600
</td>
</tr>
<tr>
<td style="text-align:left;">
rs1628530
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
104029307
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:right;">
0.1235526
</td>
</tr>
<tr>
<td style="text-align:left;">
rs17114433
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
104043015
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:left;">
A
</td>
<td style="text-align:right;">
0.0247393
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">cormat</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;            [,1]      [,2]       [,3]</span></span>
<span><span class="co">#&gt; [1,]  1.0000000 0.3862754 -0.2043553</span></span>
<span><span class="co">#&gt; [2,]  0.3862754 1.0000000  0.3080552</span></span>
<span><span class="co">#&gt; [3,] -0.2043553 0.3080552  1.0000000</span></span></code></pre></div>
<p>Finally, we generate a heatmap to visually represent the LD
estimates:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cormat</span> <span class="op">&lt;-</span> <span class="va">res</span><span class="op">$</span><span class="va">cormat</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/diag.html" class="external-link">diag</a></span><span class="op">(</span><span class="va">cormat</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="cn">NA</span></span>
<span><span class="va">cormelt</span> <span class="op">&lt;-</span> <span class="fu">reshape2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html" class="external-link">melt</a></span><span class="op">(</span><span class="va">cormat</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">cormelt</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Var1</span>, y <span class="op">=</span> <span class="va">Var2</span>, fill <span class="op">=</span> <span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_tile.html" class="external-link">geom_tile</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_fill_gradient2</a></span><span class="op">(</span>low <span class="op">=</span> <span class="st">"blue"</span>, high <span class="op">=</span> <span class="st">"red"</span>, </span>
<span>                       mid <span class="op">=</span> <span class="st">"white"</span>, </span>
<span>                       midpoint <span class="op">=</span> <span class="fl">0</span>, limit <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1</span>,<span class="fl">1</span><span class="op">)</span>, </span>
<span>                       space <span class="op">=</span> <span class="st">"Lab"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_reverse</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        axis.ticks <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        panel.grid.major <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        panel.grid.minor <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_fixed.html" class="external-link">coord_fixed</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p1</span></span></code></pre></div>
<p><img src="computeLD_example_files/figure-html/heatmap-1.png" class="r-plt" alt="" width="576"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">#ggsave(file="docs/figure/computeLD_example.pdf", p1)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ul>
<li>Lee et al. DISTMIX: direct imputation of summary statistics for
unmeasured SNPs from mixed ethnicity cohorts. Bioinformatics. <a href="https://doi.org/10.1093/bioinformatics/btv348" class="external-link uri">https://doi.org/10.1093/bioinformatics/btv348</a>.</li>
<li>Lee and Bacanu. GAUSS: a summary-statistics-based R package for
accurate estimation of linkage disequilibrium for variants, Gaussian
imputation, and TWAS analysis of cosmopolitan cohorts. Bioinformatics.
<a href="https://doi.org/10.1093/bioinformatics/btae203" class="external-link uri">https://doi.org/10.1093/bioinformatics/btae203</a>.</li>
</ul>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://github.com/dleelab" class="external-link">Donghyung Lee</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
