# Generated by using Rcpp::compileAttributes() -> do not edit by hand
# Generator token: 10BE3573-1514-4C36-9D1C-5A225CD40393

#' Calculate population weights using RAF
#' 
#' @param input_file file name of input data containing rsid, chr, bp, a1, a2, and af1 
#' @param reference_index_file file name of reference panel index data
#' @param reference_data_file  file name of reference panel data
#' @param reference_pop_desc_file file name of reference panel population description data
#' @param interval number of non-overlapping SNP sets used in calculating population weights 
#' @return R data frame containing population IDs and weights 
afmix <- function(input_file, reference_index_file, reference_data_file, reference_pop_desc_file, interval = NULL) {
    .Call(`_gauss_afmix`, input_file, reference_index_file, reference_data_file, reference_pop_desc_file, interval)
}

#' Compute LD for measured SNPs from mixed ethnicity cohorts
#' 
#' @param chr chromosome number
#' @param start_bp start base pair position of prediction window
#' @param end_bp end base pair position of prediction window
#' @param pop_wgt_df R data frame containing population IDs and weights
#' @param input_file file name of GWAS summary statistics data containing rsid, chr, bp, a1, a2, af1, and z
#' @param reference_index_file file name of reference panel index data
#' @param reference_data_file  file name of reference panel data
#' @param reference_pop_desc_file file name of reference panel population description data
#' @param af1_cutoff cutoff of reference allele, a1, frequency
#' @return R list containing a data frame containing rsid, chr, bp, a1, a2 and af1mix and a correlation matrix 
computeLD <- function(chr, start_bp, end_bp, pop_wgt_df, input_file, reference_index_file, reference_data_file, reference_pop_desc_file, af1_cutoff = NULL) {
    .Call(`_gauss_computeLD`, chr, start_bp, end_bp, pop_wgt_df, input_file, reference_index_file, reference_data_file, reference_pop_desc_file, af1_cutoff)
}

#' Calculate population weights using RAF
#' 
#' @param input_file file name of input data containing rsid, chr, bp, a1, a2, and af1 
#' @param reference_index_file file name of reference panel index data
#' @param reference_data_file  file name of reference panel data
#' @param reference_pop_desc_file file name of reference panel population description data
#' @param interval number of non-overlapping SNP sets used in calculating population weights 
#' @return R data frame containing population IDs and weights 
cpw2 <- function(input_file, reference_index_file, reference_data_file, reference_pop_desc_file, interval = NULL) {
    .Call(`_gauss_cpw2`, input_file, reference_index_file, reference_data_file, reference_pop_desc_file, interval)
}

#' Direct imputation of summary statistics for unmeasured SNPs from ethnically homogeneous cohorts
#' 
#' @param chr chromosome number
#' @param start_bp start base pair position of prediction window
#' @param end_bp end base pair position of prediction window
#' @param wing_size the size of the area flanking the left and right of the prediction window
#' @param study_pop study population group
#' @param input_file file name of GWAS summary statistics data containing rsid, chr, bp, a1, a2, and z
#' @param reference_index_file file name of reference panel index data
#' @param reference_data_file  file name of reference panel data
#' @param reference_pop_desc_file file name of reference panel population description data
#' @param af1_cutoff cutoff of reference allele, a1, frequency
#' @return R dataframe containing rsid, chr, bp, a1, a2, af1ref, z, pval, info, type   
dist <- function(chr, start_bp, end_bp, wing_size, study_pop, input_file, reference_index_file, reference_data_file, reference_pop_desc_file, af1_cutoff = NULL) {
    .Call(`_gauss_dist`, chr, start_bp, end_bp, wing_size, study_pop, input_file, reference_index_file, reference_data_file, reference_pop_desc_file, af1_cutoff)
}

#' Direct imputation of summary statistics for unmeasured SNPs from mixed ethnicity cohorts
#' 
#' @param chr chromosome number
#' @param start_bp start base pair position of prediction window
#' @param end_bp end base pair position of prediction window
#' @param wing_size the size of the area flanking the left and right of the prediction window
#' @param pop_wgt_df R data frame containing population IDs and weights
#' @param input_file file name of GWAS summary statistics data containing rsid, chr, bp, a1, a2, af1, and z
#' @param reference_index_file file name of reference panel index data
#' @param reference_data_file  file name of reference panel data
#' @param reference_pop_desc_file file name of reference panel population description data
#' @param af1_cutoff cutoff of reference allele, a1, frequency
#' @return R data frame containing rsid, chr, bp, a1, a2, af1mix, z, pval, info, type 
distmix <- function(chr, start_bp, end_bp, wing_size, pop_wgt_df, input_file, reference_index_file, reference_data_file, reference_pop_desc_file, af1_cutoff = NULL) {
    .Call(`_gauss_distmix`, chr, start_bp, end_bp, wing_size, pop_wgt_df, input_file, reference_index_file, reference_data_file, reference_pop_desc_file, af1_cutoff)
}

#' Joint effect on phenotype of eQTLs/functional SNPs associated with a gene
#' 
#' @param study_pop study population group
#' @param input_file file name of GWAS summary statistics data containing rsid, chr, bp, a1, a2 and z
#' @param annotation file name of the SNP annotation data set 
#' @param reference_index_file file name of reference panel index data
#' @param reference_data_file  file name of reference panel data
#' @param reference_pop_desc_file file name of reference panel population description data
#' @param af1_cutoff cutoff of reference allele, a1, frequency
#' @return R dataframe containing geneid, chisq, df, jepeg_pval, num_snp, top_categ, top_categ_pval, top_snp, and top_snp_pval   
jepeg <- function(study_pop, input_file, annotation_file, reference_index_file, reference_data_file, reference_pop_desc_file, af1_cutoff = NULL) {
    .Call(`_gauss_jepeg`, study_pop, input_file, annotation_file, reference_index_file, reference_data_file, reference_pop_desc_file, af1_cutoff)
}

#' JEPEGMIX: gene-level joint analysis of functional SNPs in cosmopolitan cohorts
#' 
#' @param pop_wgt_df R data frame containing population IDs and weights
#' @param input_file file name of GWAS summary statistics data containing rsid, chr, bp, a1, a2 and z
#' @param annotation file name of the SNP annotation data set 
#' @param reference_index_file file name of reference panel index data
#' @param reference_data_file  file name of reference panel data
#' @param reference_pop_desc_file file name of reference panel population description data
#' @param af1_cutoff cutoff of reference allele, a1, frequency
#' @return R dataframe containing geneid, chisq, df, jepeg_pval, num_snp, top_categ, top_categ_pval, top_snp, and top_snp_pval   
jepegmix <- function(pop_wgt_df, input_file, annotation_file, reference_index_file, reference_data_file, reference_pop_desc_file, af1_cutoff = NULL) {
    .Call(`_gauss_jepegmix`, pop_wgt_df, input_file, annotation_file, reference_index_file, reference_data_file, reference_pop_desc_file, af1_cutoff)
}

#' Prepare datasets for QCAT analysis
#' 
#' @param chr chromosome number
#' @param start_bp start base pair position of prediction window
#' @param end_bp end base pair position of prediction window
#' @param wing_size the size of the area flanking the left and right of the prediction window
#' @param study_pop study population group
#' @param input_file file name of GWAS summary statistics data containing rsid, chr, bp, a1, a2, af1, and z
#' @param reference_index_file file name of reference panel index data
#' @param reference_data_file  file name of reference panel data
#' @param reference_pop_desc_file file name of reference panel population description data
#' @param af1_cutoff cutoff of reference allele, a1, frequency
#' @return R dataframe containing rsid, chr, bp, a1, a2, af1ref, z, qcat_chisq, qcat_pval, type
prep_qcat <- function(chr, start_bp, end_bp, wing_size, study_pop, input_file, reference_index_file, reference_data_file, reference_pop_desc_file, af1_cutoff = NULL) {
    .Call(`_gauss_prep_qcat`, chr, start_bp, end_bp, wing_size, study_pop, input_file, reference_index_file, reference_data_file, reference_pop_desc_file, af1_cutoff)
}

#' Prepare datasets for Recessive Imputation
#' 
#' @param chr chromosome number
#' @param start_bp start base pair position of prediction window
#' @param end_bp end base pair position of prediction window
#' @param wing_size the size of the area flanking the left and right of the prediction window
#' @param pop_wgt_df R data frame containing population IDs and weights
#' @param input_file file name of GWAS summary statistics data containing rsid, chr, bp, a1, a2, af1, and z
#' @param reference_index_file file name of reference panel index data
#' @param reference_data_file  file name of reference panel data
#' @param reference_pop_desc_file file name of reference panel population description data
#' @param af1_cutoff cutoff of reference allele, a1, frequency
#' @return A List containing:
#'         - snplist: A data frame of SNPs in the prediction window with columns rsid, chr, bp, a1, a2, af1mix, z, and type,
#'         - zvec: A numeric vector of Z-scores for measured SNPs in the extended window,
#'         - cormat: A correlation matrix (additive-coded) among measured SNPs in the extended window,
#'         - cormat_add: A correlation matrix between additive-coded predicted SNPs and additive-coded measured SNPs,
#'         - cormat_dom: A correlation matrix between dominant-coded predicted SNPs and additive-coded measured SNPs,
#'         - cormat_rec: A correlation matrix between recessive-coded predicted SNPs and additive-coded measured SNPs.
NULL

#' Prepare datasets for QCATMIX analysis
#' 
#' @param chr chromosome number
#' @param start_bp start base pair position of prediction window
#' @param end_bp end base pair position of prediction window
#' @param wing_size the size of the area flanking the left and right of the prediction window
#' @param pop_wgt_df R data frame containing population IDs and weights
#' @param input_file file name of GWAS summary statistics data containing rsid, chr, bp, a1, a2, af1, and z
#' @param reference_index_file file name of reference panel index data
#' @param reference_data_file  file name of reference panel data
#' @param reference_pop_desc_file file name of reference panel population description data
#' @param af1_cutoff cutoff of reference allele, a1, frequency
#' @return R dataframe containing rsid, chr, bp, a1, a2, af1mix, z, qcat_chisq, qcat_pval, type
prep_qcatmix <- function(chr, start_bp, end_bp, wing_size, pop_wgt_df, input_file, reference_index_file, reference_data_file, reference_pop_desc_file, af1_cutoff = NULL) {
    .Call(`_gauss_prep_qcatmix`, chr, start_bp, end_bp, wing_size, pop_wgt_df, input_file, reference_index_file, reference_data_file, reference_pop_desc_file, af1_cutoff)
}

prep_recessive_impute <- function(chr, start_bp, end_bp, wing_size, pop_wgt_df, input_file, reference_index_file, reference_data_file, reference_pop_desc_file, af1_cutoff = NULL) {
    .Call(`_gauss_prep_recessive_impute`, chr, start_bp, end_bp, wing_size, pop_wgt_df, input_file, reference_index_file, reference_data_file, reference_pop_desc_file, af1_cutoff)
}

#' Testing causality of variants from ethnically homogeneous cohorts
#' 
#' @param chr chromosome number
#' @param start_bp start base pair position of prediction window
#' @param end_bp end base pair position of prediction window
#' @param wing_size the size of the area flanking the left and right of the prediction window
#' @param study_pop study population group
#' @param input_file file name of GWAS summary statistics data containing rsid, chr, bp, a1, a2, af1, and z
#' @param reference_index_file file name of reference panel index data
#' @param reference_data_file  file name of reference panel data
#' @param reference_pop_desc_file file name of reference panel population description data
#' @param af1_cutoff cutoff of reference allele, a1, frequency
#' @return R dataframe containing rsid, chr, bp, a1, a2, af1ref, z, qcat_chisq, qcat_pval, type
qcat <- function(chr, start_bp, end_bp, wing_size, study_pop, input_file, reference_index_file, reference_data_file, reference_pop_desc_file, af1_cutoff = NULL) {
    .Call(`_gauss_qcat`, chr, start_bp, end_bp, wing_size, study_pop, input_file, reference_index_file, reference_data_file, reference_pop_desc_file, af1_cutoff)
}

#' Testing causality of variants from mixed ethnicity cohorts
#' 
#' @param chr chromosome number
#' @param start_bp start base pair position of prediction window
#' @param end_bp end base pair position of prediction window
#' @param wing_size the size of the area flanking the left and right of the prediction window
#' @param pop_wgt_df R data frame containing population IDs and weights
#' @param input_file file name of GWAS summary statistics data containing rsid, chr, bp, a1, a2, af1, and z
#' @param reference_index_file file name of reference panel index data
#' @param reference_data_file  file name of reference panel data
#' @param reference_pop_desc_file file name of reference panel population description data
#' @param af1_cutoff cutoff of reference allele, a1, frequency
#' @return R dataframe containing rsid, chr, bp, a1, a2, af1mix, z, qcat_chisq, qcat_pval, type
qcatmix <- function(chr, start_bp, end_bp, wing_size, pop_wgt_df, input_file, reference_index_file, reference_data_file, reference_pop_desc_file, af1_cutoff = NULL) {
    .Call(`_gauss_qcatmix`, chr, start_bp, end_bp, wing_size, pop_wgt_df, input_file, reference_index_file, reference_data_file, reference_pop_desc_file, af1_cutoff)
}

#' Simulate LD for user provided SNPs from mixed ethnicity cohorts
#' 
#' @param chr chromosome number
#' @param start_bp start base pair position of prediction window
#' @param end_bp end base pair position of prediction window
#' @param pop_wgt_df R data frame containing population IDs and weights
#' @param sim_size Number of simulated subjects
#' @param input_file file name of GWAS summary statistics data containing rsid, chr, bp, a1, a2, af1, and z
#' @param reference_index_file file name of reference panel index data
#' @param reference_data_file  file name of reference panel data
#' @param reference_pop_desc_file file name of reference panel population description data
#' @param af1_cutoff cutoff of reference allele, a1, frequency
#' @return R list containing a data frame containing rsid, chr, bp, a1, a2 and af1mix and a correlation matrix 
simulateLD <- function(chr, start_bp, end_bp, pop_wgt_df, sim_size, input_file, reference_index_file, reference_data_file, reference_pop_desc_file, af1_cutoff = NULL) {
    .Call(`_gauss_simulateLD`, chr, start_bp, end_bp, pop_wgt_df, sim_size, input_file, reference_index_file, reference_data_file, reference_pop_desc_file, af1_cutoff)
}

#' Calculate population weights using association Z-scores
#' 
#' This function first finds ancestry-informed SNPs by computing a normalized variance,
#' ratio = var(af1)/(mean(af1)*(1-mean(af1))) and choose top 1% of SNPs with highest 
#' ratio and use these SNPs in the zmix_prep procedure.
#' It uses an interval parameter to determine how SNPs are selected for analysis. 
#' If interval is not provided, it defaults to 1, meaning every SNP is considered.
#' SNP Pairing: Pairs each SNP with every other SNP in the subvector (snp_subvec)
#' If Interval=1000, it pairs SNP_0 with SNP_1000, SNP_0 with SNP_2000 ... and SNP_1000
#' with SNP_2000, SNP_1000 with SNP_3000 ...
#' 
#' @param input_file file name of input data containing rsid, chr, bp, a1, a2, and z 
#' @param reference_index_file file name of reference panel index data
#' @param reference_data_file  file name of reference panel data
#' @param reference_pop_desc_file file name of reference panel population description data
#' @param interval stepping distance within the SNP vector for selecting the first SNP of each pair. 
#' @param percentile percentile cutoff for normalized variance
#' @return R data frame containing population IDs and weights 
prep_zmix5 <- function(input_file, reference_index_file, reference_data_file, reference_pop_desc_file, percentile = NULL, interval = NULL) {
    .Call(`_gauss_prep_zmix5`, input_file, reference_index_file, reference_data_file, reference_pop_desc_file, percentile, interval)
}

#' Calculate population weights using association Z-scores
#' 
#' ZMIX4 employs a more selective approach by using a fixed offset to 
#' determine the SNP pairs. For instance, if the offset is set to 3 and 
#' the interval is specified as 1000, the function will pair the first SNP 
#' (SNP_0) with the fourth one (SNP_3), then SNP_1000 with SNP_1003, and so forth. 
#' In subsequent iterations, this pattern continues in a similar manner - 
#' SNP_1 gets paired with SNP_4, SNP_1001 with SNP_1004, and so on, 
#' effectively creating pairs at specific intervals with a consistent offset.
#' 
#' @param input_file file name of input data containing rsid, chr, bp, a1, a2, and z 
#' @param reference_index_file file name of reference panel index data
#' @param reference_data_file  file name of reference panel data
#' @param reference_pop_desc_file file name of reference panel population description data
#' @param interval stepping distance within the SNP vector for selecting the first SNP of each pair. 
#' @param offset distance between the two SNPs in a pair
#' @return R data frame containing population IDs and weights 
prep_zmix4 <- function(input_file, reference_index_file, reference_data_file, reference_pop_desc_file, interval = NULL, offset = NULL) {
    .Call(`_gauss_prep_zmix4`, input_file, reference_index_file, reference_data_file, reference_pop_desc_file, interval, offset)
}

#' Calculate population weights using association Z-scores
#' 
#' Pairing Logic Based on Steps: This function pairs each SNP with a 
#' specific number of subsequent SNPs as defined by the 'steps' parameter. 
#' This function is a variant of ZMIX. While ZMIX2 pairs one SNP in the snp_vec
#' with all the remaining SNPs, ZMIX3 limits the number of SNPs to be paired by 
#' using the 'steps' argument. For example, if 'steps' is set to 5, 
#' each SNP in the snp_subvec is paired with its next 5 subsequent SNPs.
#' 
#' @param input_file file name of input data containing rsid, chr, bp, a1, a2, and z 
#' @param reference_index_file file name of reference panel index data
#' @param reference_data_file  file name of reference panel data
#' @param reference_pop_desc_file file name of reference panel population description data
#' @param interval stepping distance within the SNP vector for selecting the first SNP of each pair. 
#' @param steps number of subsequent SNPs to pair with each SNP.
#' @return R data frame containing population IDs and weights 
prep_zmix3 <- function(input_file, reference_index_file, reference_data_file, reference_pop_desc_file, interval = NULL, steps = NULL) {
    .Call(`_gauss_prep_zmix3`, input_file, reference_index_file, reference_data_file, reference_pop_desc_file, interval, steps)
}

#' Calculate population weights using association Z-scores
#' 
#' Interval and Offset Logic: Instead of considering all pairwise combinations, 
#' zmix2 selects SNP pairs based on a fixed offset. For example, if offset is 3 and
#' and interval is 1000, it pairs SNP_0 with SNP_3, SNP_1000 with SNP_1003 ...
#' Reduced Pairing Scope: This approach reduces the total number of SNP pairs 
#' considered compared to zmix, focusing on pairs separated by a specific distance 
#' (offset) in the SNP vector.
#' 
#' @param input_file file name of input data containing rsid, chr, bp, a1, a2, and z 
#' @param reference_index_file file name of reference panel index data
#' @param reference_data_file  file name of reference panel data
#' @param reference_pop_desc_file file name of reference panel population description data
#' @param interval stepping distance within the SNP vector for selecting the first SNP of each pair. 
#' @param offset distance between the two SNPs in a pair
#' @return R data frame containing population IDs and weights 
prep_zmix2 <- function(input_file, reference_index_file, reference_data_file, reference_pop_desc_file, interval = NULL, offset = NULL) {
    .Call(`_gauss_prep_zmix2`, input_file, reference_index_file, reference_data_file, reference_pop_desc_file, interval, offset)
}

#' Calculate population weights using association Z-scores
#' 
#' It uses an interval parameter to determine how SNPs are selected for analysis. 
#' If interval is not provided, it defaults to 1, meaning every SNP is considered.
#' SNP Pairing: Pairs each SNP with every other SNP in the subvector (snp_subvec)
#' If Interval=1000, it pairs SNP_0 with SNP_1000, SNP_0 with SNP_2000 ... and SNP_1000
#' with SNP_2000, SNP_1000 with SNP_3000 ...
#' 
#' @param input_file file name of input data containing rsid, chr, bp, a1, a2, and z 
#' @param reference_index_file file name of reference panel index data
#' @param reference_data_file  file name of reference panel data
#' @param reference_pop_desc_file file name of reference panel population description data
#' @param interval stepping distance within the SNP vector for selecting the first SNP of each pair. 
#' @return R data frame containing population IDs and weights 
prep_zmix <- function(input_file, reference_index_file, reference_data_file, reference_pop_desc_file, interval = NULL) {
    .Call(`_gauss_prep_zmix`, input_file, reference_index_file, reference_data_file, reference_pop_desc_file, interval)
}

